File: mkblog-spring-cache-redis/src/main/java/com/mustafakoroglu/blog/scr/rest/CustomerController.java
package com.mustafakoroglu.blog.scr.rest;

import com.mustafakoroglu.blog.scr.customer.Customer;
import com.mustafakoroglu.blog.scr.customer.CustomerService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * @author Mustafa Köroğlu
 */
@RestController
@RequestMapping("/customers")
public class CustomerController {
    private final CustomerService customerService;

    public CustomerController(CustomerService customerService) {
        this.customerService = customerService;
    }

    @GetMapping
    List<Customer> getCustomers() {
        return customerService.getCustomers();
    }

    @PostMapping
    void saveCustomer(Customer customer) {
        customerService.save(customer);
    }
}

File: mkblog-thymeleaf-layout-dialect/src/main/kotlin/com/mustafakoroglu/blog/mkblogthymeleaflayoutdialect/controller/CustomerController.kt
package com.mustafakoroglu.blog.mkblogthymeleaflayoutdialect.controller

import com.mustafakoroglu.blog.mkblogthymeleaflayoutdialect.domain.Customer
import com.mustafakoroglu.blog.mkblogthymeleaflayoutdialect.repository.CustomerRepository
import org.springframework.stereotype.Controller
import org.springframework.ui.ModelMap
import org.springframework.web.bind.annotation.*

/**
 * @author Mustafa Köroğlu <mustafa@koroglu.email>
 */
@Controller
@RequestMapping("/customers")
class CustomerController(val customerRepository: CustomerRepository) {

    @GetMapping
    fun listPage(map: ModelMap): String {
        map.put("customers", customerRepository.findAll())
        return "customer_list"
    }

    @GetMapping("/new")
    fun newCustomerPage(map: ModelMap): String {
        map.put("customer", Customer())
        return "new_customer"
    }

    @PostMapping
    fun addCustomer(@ModelAttribute customer: Customer): String {
        customerRepository.save(customer)
        return "redirect:/customers"
    }

    @GetMapping("/{customer}/delete")
    fun deleteCustomer(@PathVariable customer: Customer): String {
        customerRepository.delete(customer)
        return "redirect:/customers"
    }
}